native_client/javascript/README.md

```diff
@@ -1 +1,29 @@
 Full project description and documentation on GitHub: [https://github.com/mozilla/DeepSpeech](https://github.com/mozilla/DeepSpeech).
+
+## Generating TypeScript Type Definitions
+
+You can generate the TypeScript type declaration file using `dts-gen`.
+This requires a compiled/installed version of the DeepSpeech NodeJS client.
+
+To generate a new `index.d.ts` type declaration file, run:
+
+```sh
+npm install -g dts-gen
+dts-gen --module deepspeech --file index.d.ts
+```
+
+### Example usage
+
+```javascript
+/// index.ts
```

Comment:
Should we write a `index.ts` next to our current `index.js` ? How do we package that and run it as typescript ?

Reply:
Ok, so we can use `tsc` after installing `typescript` NPM package.

Reply:
> Ok, so we can use `tsc` after installing `typescript` NPM package.

Hm that is just re-generating my `index.js`

Reply:
My understanding is that `index.d.js` allows our package to support usage from both JavaScript and TypeScript, so having `index.d.js` means an `index.ts` is not needed.